internal_ws/ykshim/src/test_api.rs

```diff
@@ -104,9 +104,8 @@ unsafe extern "C" fn __ykshimtest_tracecompiler_local_dead(tc: *mut TraceCompile
 
 /// Find a symbol's address in the current memory image. Returns NULL if it can't be found.
```

Comment:
We need to document explicitly that this consumes a `CString` to maintain the guarantee in https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_raw. I did wonder if we could even make the input type by `CString` but I guess that doesn't work across the boundary?

Reply:
Actually, there's no need for this to consume the argument. I'll see if I can make that work.

Reply:
bd718ec shows what it would take to make this function (and 2 others) not consume their argument.

I'm not very sure which approach is best. Do you have a preference? How hard should we try to avoid consumption?

Reply:
(there's a bug, i missed an `as_ptr()`, but you get the idea -- as_ptr() vs. consumption with documentation).

Reply:
I think this is is better, because not only do we not have to consume things, we don't even have to rely on the various restrictions of `CString`.

Reply:
Oops, "this" is ambiguous: I meant that bd718ec (with your suggested bug fix!) is better.